canal/canal.go

```diff
@@ -197,8 +218,40 @@ func (c *Canal) Ctx() context.Context {
 	return c.ctx
 }
 
+func (c *Canal) checkTableMatch(key string) bool {
+	// no filter, return true
+	if c.tableMatchCache == nil {
+		return true
+	}
+	// cache hit
+	if rst, ok := c.tableMatchCache[key]; ok {
```

Comment:
please use a lock to operate the cache map.

Reply:
OK, because tableMatchCache only be used in GetTable, so I use tableLock instead,.